protocol/lavasession/provider_session_manager_test.go

```diff
@@ -177,9 +184,83 @@ func TestHappyFlowBadgePSM(t *testing.T) {
 	require.Equal(t, sps.PairingEpoch, epoch1)
 }
 
+func TestHappyFlowBadgePSMMultipleRoutines(t *testing.T) {
```

Comment:
This test is added for this task: https://lavanet.atlassian.net/browse/PRT-712?atlOrigin=eyJpIjoiYjFmODM3ZDEzMWNmNDQ3OWExZDY5NTY2ZWNmMDdlYzYiLCJwIjoiaiJ9